### ISSUE TITLE
'console' does not support string substitution

### ISSUE THREAD

**Issue by gear54rus** [scriptish/scriptish#251]
Hello,

I've been using Scriptish for quite some time and previously also used Greasemonkey. I have to say: awesome job! Exactly the improvement that is expected from GM.

Here's the issue I've found:

There exists a following spec: https://developer.mozilla.org/en-US/docs/Web/API/console#Using_string_substitutions

Which dictates that we can use various `console` methods (`log`, `info`, etc.) the following way:

``` javascript
console.log('string with %o some %d tokens', {a:10}, 5);
```

which will output:

![2](https://cloud.githubusercontent.com/assets/8073599/3890295/b1abedcc-221c-11e4-9154-ae14e1b51cc5.png)

However, this does not work in Scriptish:

The following code:

``` javascript
console.log('test %o %d test', {a:10}, 5);
unsafeWindow.console.log('test %o %d test', {a:10}, 5);
console.log.apply(['test %o %d test', {a:10}, 5]);
unsafeWindow.console.log.apply(['test %o %d test', {a:10}, 5]);
```

will output (_order is messed up for some reason, refer to line numbers on the right_):

![1](https://cloud.githubusercontent.com/assets/8073599/3890351/44162f7e-221d-11e4-8ddb-ddaeaa1893fa.png)

Here, one can see the difference between built-in `console` and the one provided by Scriptish.

One could argue that we could use built-in `console` in specific cases but unfortunately one cannot call `apply` on `console.method` (as shown in example): `[1,2,3].forEach(console.log)` will show error. Scriptish's `console` can be `apply`'d - major advantage.

Thanks!

**Comment by Mark2834** [comment]
Substitution works if I use regular scripting on a web page, but in no way can I get the same code to work from a userscript run by Scriptish.  Do you know whether it works with Greasemonkey?
GM_log doesn't appear to support that feature either.

**Comment by gear54rus** [comment]
If you wanted to, you could call original console.\* functions through unsafeWindow (don't forget to **@grant unsafeWindow**) and they will work just fine 100% identical to what you see in non-scriptish environment. This doesn't work for my case though and that's why I posted this.

TL;DR: try using `unsafeWindow.console.*`

**Comment by gear54rus** [comment]
I now realized that I've been stupid and that one could use **console.log.apply(console, []);** (not being able to **apply** was why I created this) to use apply... Still, the issue itself exists...

**Comment by erikvold** [comment]
I'm going to archive the repo soon.
